Update for NativeScript 3.0-rc

DickSmith opened the pull request:
**Configured as detailed here:**
[NativeScript v3.0 Breaking Changes](https://github.com/NativeScript/NativeScript/blob/master/Modules30Changes.md#property-system)

**Notes:**
1. _nativeView field removed since nativeView is now part of the API.
2. Stricter types for properties.
3. Otherwise should be backwards compatible with standard templates/css usage. (I didn't need to change any of my own or the demo templates/css.)
4. Tested with both Vanilla NativeScript Demo (.ts minimally updated per 3.0 changes) and with my own NS+ng project.

Obviously may be some changes before 3.0-release, but should remain close enough since I followed the example in the change log as well as other examples in the master repo, and works for my current project being built against Nativescript 3.0-rc and nativescript-angular 2.0-rc.

Commit:
Update for NativeScript 3.0-rc
Configured as detailed here:
https://github.com/NativeScript/NativeScript/blob/master/Modules30Changes.md#property-system

Notes:
_nativeView field removed since nativeView is now part of the API.
Tested with both Vanilla NativeScript Demo (minimally updated per 3.0 changes) and with my own NS+ng project.

NathanWalker commented:
@DickSmith Thank you! lgtm!!
@bradmartin I'm on fence with some other plugins if we go ahead and release major point updates with 3.0 compatibility or wait til final release. We could just bump to 2.0.0 with this.

bradmartin commented:
Definitely a major bump with the breaking changes. I'm fine with going ahead and updating it - just need a notice on the README about < 3.0 using our last published release.

Commit:
Versioning
Per the discussion:
1. Bumped the major version and added prerelease (just to make it clear that these changes are based are based on **3.0-rc** and could still be subject to change; although hopefully not).
2. Added `tns-core-modules` to peer dependencies to provide warnings during `nom install` if **<3.x**.
3. Updated README to provide info on installation for both **NS 2.x** and **NS 3+** projects.

DickSmith commented:
Per the discussion:
1. Bumped the major version and added prerelease (just to make it clear that these changes are based on **3.0-rc** and could still be subject to change; although hopefully not).
2. Added `tns-core-modules` to peer dependencies to provide warnings during `nom install` if **<3.x**.
3. Updated README to provide info on installation for both **NS 2.x** and **NS 3+** projects.

NathanWalker commented:
This is excellent @DickSmith merging and will bump per discussion shortly. 👍

NathanWalker commented:
@DickSmith This is now published as `2.0.0-rc.0` please give it a try and let us know if it works with your 3.0 project 👍

DickSmith commented:
So far so good. Using the `2.0.0-rc.0` from npm and it's working perfectly for several different components we use CardView for in our project.
